File: frontend/src/db/models/Note.ts
import { Model } from '@nozbe/watermelondb'
import { field, date, relation } from '@nozbe/watermelondb/decorators'
import Session from './Session'

export default class Note extends Model {
  static table = 'notes'
  static associations = {
    sessions: { type: 'belongs_to' as const, key: 'session_id' },
  }

  @field('session_id')   sessionId!: string
  @field('server_id')    serverId!: string | null
  @field('doing_now')    doingNow!: string | null
  @field('next_step')    nextStep!: string | null
  @field('open_thought') openThought!: string | null
  @field('audio_url')    audioUrl!: string | null
  @field('ai_summary')   aiSummary!: string | null
  @date('created_at') createdAt!: Date
  @field('synced') synced!: boolean
  @field('updated_at') updatedAt!: Date

  @relation('sessions', 'session_id') session!: Session
}

File: backend/app/models/note.py
from sqlalchemy import Column, String, DateTime, ForeignKey, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
import uuid
from app.database import Base

class Note(Base):
    __tablename__ = "notes"

    id           = Column(String, primary_key=True, default=lambda: str(uuid.uuid4()))
    session_id   = Column(String, ForeignKey("sessions.id"), nullable=False)

    # Campos do modal de anotação
    doing_now    = Column(Text, nullable=True)
    next_step    = Column(Text, nullable=True)
    open_thought = Column(Text, nullable=True)

    # Áudio e IA
    audio_url    = Column(String, nullable=True)   # caminho do arquivo gravado
    ai_summary   = Column(Text, nullable=True)     # retorno processado pela IA

    # Sync com WatermelonDB
    watermelon_id = Column(String, nullable=True, unique=True)
    created_at    = Column(DateTime(timezone=True), server_default=func.now())
    updated_at    = Column(DateTime(timezone=True), onupdate=func.now(),
                           server_default=func.now())

    session = relationship("Session", back_populates="notes")